src/Shared/Recruit.Vacancies.Client/Infrastructure/ReferenceData/BankHolidays/BankHolidayProvider.cs

```diff
@@ -11,19 +12,28 @@ namespace Esfa.Recruit.Vacancies.Client.Infrastructure.Services
     public class BankHolidayProvider : IBankHolidayProvider
     {
         private readonly IReferenceDataReader _referenceDataReader;
+        private readonly ICache _cache;
 
-        public BankHolidayProvider(IReferenceDataReader referenceDataReader)
+        private DateTime CacheAbsoluteExpiryTime => DateTime.UtcNow.Date.AddDays(1);
+
+        public BankHolidayProvider(IReferenceDataReader referenceDataReader, ICache cache)
         {
             _referenceDataReader = referenceDataReader;
+            _cache = cache;
         }
 
         public async Task<List<DateTime>> GetBankHolidaysAsync()
         {
-            var bankHolidayReferenceData = await _referenceDataReader.GetReferenceData<BankHolidays>();
+            return await _cache.CacheAsideAsync(CacheKeys.BankHolidays,
+                CacheAbsoluteExpiryTime,
+                async () =>
```

Comment:
On a couple of the providers I think we have some inconsistency in terms of the null checks and logging.